src/ILLink.Tasks/build/Microsoft.NET.ILLink.targets

```diff
@@ -279,9 +279,9 @@ Copyright (c) .NET Foundation. All rights reserved.
       <ManagedAssemblyToLink Include="@(_TrimmableManagedAssemblyToLink)" IsTrimmable="true" />
     </ItemGroup>
 
-    <!-- Root the main assembly, whether or not it has IsTrimmable set. -->
+    <!-- Root the main assembly entry point. -->
     <ItemGroup>
-      <TrimmerRootAssembly Include="@(IntermediateAssembly)" />
+      <TrimmerRootAssembly Include="@(IntermediateAssembly)" RootMode="EntryPoint" />
```

Comment:
@sbomer for Android apps are actually class libraries, there is no `static void Main()` because Android's entry point is either an `Android.App.Application` or `Android.App.Activity`.

I think this change causes:
```xml
ILLink error IL1034: Root assembly 'MyAndroidApp, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null' does not have entry point.
```

Should I do this in the Android workload?
```
<ItemGroup>
  <TrimmerRootAssembly Update="@(TrimmerRootAssembly)" RootMode="Library" />
</ItemGroup>
```